src/components/asset-list/RecyclerAssetList2/core/getLayoutProvider.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  Dimension,
+  Layout,
+  LayoutManager,
+  LayoutProvider,
+} from 'recyclerlistview';
+import ViewDimensions from './ViewDimensions';
+import { BaseCellType, CellType } from './ViewTypes';
+import { deviceUtils } from '@rainbow-me/utils';
+
+const getStyleOverridesForIndex = (indices: number[]) => (index: number) => {
+  if (indices.includes(index)) {
+    return {
+      zIndex: 1000,
+    };
+  }
+  return undefined;
+};
+
+class BetterLayoutProvider extends LayoutProvider {
+  shouldRefreshWithAnchoring = false;
+  private readonly indicesToOverride: number[];
+  constructor(
+    getLayoutTypeForIndex: (index: number) => string | number,
+    setLayoutForType: (
+      type: string | number,
+      dim: Dimension,
+      index: number
+    ) => void,
+    indicesToOverride: number[]
+  ) {
+    super(getLayoutTypeForIndex, setLayoutForType);
+    this.indicesToOverride = indicesToOverride;
+  }
+  public newLayoutManager(
+    renderWindowSize: Dimension,
+    isHorizontal?: boolean,
+    cachedLayouts?: Layout[]
+  ): LayoutManager {
+    const oldLayoutManager = super.newLayoutManager(
+      renderWindowSize,
+      isHorizontal,
+      cachedLayouts
+    );
+    oldLayoutManager.getStyleOverridesForIndex = getStyleOverridesForIndex(
+      this.indicesToOverride
+    );
+    return oldLayoutManager;
+  }
+}
+
+const getLayoutProvider = (briefSectionsData: BaseCellType[]) => {
+  const indicesToOverride = briefSectionsData
+    .map((val, index) => {
+      if (
+        val.type === CellType.ASSETS_HEADER ||
+        val.type === CellType.NFTS_HEADER
+      ) {
+        return { index };
+      }
+      return undefined;
+    })
+    .filter(Boolean)
+    .map(wrapped => wrapped?.index) as number[];
```

Comment:
Why don't we use single for loop here? 
Instead of returning object and undefined, then filtering undefined stuff and then returning indexes we can just push indexes to array

Reply:
we can